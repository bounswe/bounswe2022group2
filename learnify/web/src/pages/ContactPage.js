import React, { useCallback, useEffect, useState } from 'react';
import './style.css'
import NavBar from '../components/NavBar';
import Footer from '../components/Footer';
import people from '../images/leutee.png';

function ContactPage() {
    const [name, setName] = useState("");
    const [description, setDescription] = useState("");
    const onSubmit = (event) => {
        event.preventDefault();
        console.log(name)
        console.log(description)
    }
    return (
        <div>
            <NavBar/>
            <div className='contact-page'>
                <div className='contact-page-left'>
                    <div className='space-20' />
                    <div className='space-10' />
                    <div className='contact-title2'>
                        About this page and issue forms:
                    </div>
                    <div className='space-10' />
                    <div className='box-contact'>
                        <div className='contact-text2'>
                            This page is for reporting any issues you may have with either the website, or the Learning Spaces themselves. 
                        </div>
                        <div className='contact-text2'>
                            You can report about:
                        </div>
                        <div className='space-3' />
                        <div className='contact-text2'>
                            - Learning spaces: Participants may have issues with the learning spaces themselves, such as technical issues with the platform or difficulty accessing certain resources. Creating an issue can help bring these problems to the attention of the team responsible for managing the learning space, so that they can be addressed and resolved.
                        </div>
                        <div className='space-3' />
                        <div className='contact-text2'>
                            - Comments or posts: Participants may have concerns about comments or posts made by other users in the learning space. These could be inappropriate or offensive comments, or posts that are not relevant to the topic being discussed.
                        </div>
                        <div className='space-3' />
                        <div className='contact-text2'>
                            - UI: Participants may have suggestions or issues with the user interface (UI) of the learning space. They may find certain features confusing or hard to use, or have suggestions for improvements that could make the learning space more user-friendly.                         </div>
                    </div>
                </div>
                <div className='contact-page-middle'>
                    <div className='space-20' />
                    <div className='space-10' />
                    <div className='contact-title'>
                        Tell Us About Any Issues
                    </div>
                    <div className='space-20' />
                    <label className='ls-box-lb-option'> Subject of your issue: </label>
                    <div className='space-12' />
                    <label>
                        <span className='contact-box' role='textbox' value='nameMessage' contentEditable='true' onInput={(e) => setName(e.target.textContent)}></span>
                    </label>
                    <div className='space-40' />
                    <label className='ls-box-lb-option'> Can you please specify the problem? </label>
                    <div className='space-12' />
                    <label>
                        <span className='contact-box2' role='textbox' value='nameMessage' contentEditable='true' onInput={(e) => setDescription(e.target.textContent)}></span>
                    </label>
                    <div className='space-5' />
                    <label>
                        <input 
                            type='submit' 
                            className='ls-create-btn' 
                            value='Submit Issue'
                            onClick={onSubmit}                 
                        />
                    </label>
                </div>
                <div className='contact-page-right'>
                    <div className='space-100' />
                    <div className='left-aligned-img'>
                        <img src={people} alt="Contact Page Illustration" height={250} />
                    </div>
                    <div className='contact-text2'>
                      "It is important for our co-learners to have a good and respectful experience in order to feel valued, motivated, and supported in their learning journey."
                    </div>
                    <div className='space-8' />
                    <div className='contact-text3'>
                        - Team Learnify.
                    </div>
                </div>
            </div>
            <Footer/>
        </div>
    )

}

export default ContactPage;