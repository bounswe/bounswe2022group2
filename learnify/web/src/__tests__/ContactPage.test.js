import React from "react";
import ContactPage from "./../pages/ContactPage";
import { screen, render, fireEvent } from "@testing-library/react";
import renderer from 'react-test-renderer';
import { BrowserRouter } from "react-router-dom";

describe("ContactPage", () => {
    test('should render page layout correctly', async () => {
        const component = renderer.create(<BrowserRouter><ContactPage /></BrowserRouter>).toJSON();
        expect(component).toMatchSnapshot();
    });
}  );

