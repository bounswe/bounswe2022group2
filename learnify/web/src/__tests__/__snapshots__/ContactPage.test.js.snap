// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactPage should render page layout correctly 1`] = `
<div>
  <nav
    className="nav"
    data-testid="navbar"
  >
    <a
      href="/home"
    >
      <img
        alt="Learnify Logo Navbar"
        height={60}
        src="logo-dblue.png"
      />
    </a>
    <ul>
      <div
        className="relative"
      >
        <input
          className="search-input-field"
          onChange={[Function]}
          placeholder="Search learning spaces"
          size={30}
          type="text"
          value=""
        />
        <div
          className="search-results-container"
        />
      </div>
      <li>
        <a
          className="navBarText"
          href="/notifications"
        >
          <img
            alt="Notifications Icon"
            height={20}
            src="notification-icon.svg"
          />
        </a>
      </li>
      <li>
        <a
          className="navBarText"
          href="/home"
        >
          Home
        </a>
      </li>
      <li>
        <a
          className="navBarText"
          href="/categories"
        >
          Categories
        </a>
      </li>
      <li>
        <a
          className="navBarText"
          href="/about"
        >
          About
        </a>
      </li>
      <li>
        <a
          className="navBarText"
          href="/contact"
        >
          Contact
        </a>
      </li>
      <li>
        <a
          className="navBarText"
          href="/profile/null"
        >
          Profile
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="contact-page"
  >
    <div
      className="contact-page-left"
    >
      <div
        className="space-20"
      />
      <div
        className="space-10"
      />
      <div
        className="contact-title2"
      >
        About this page and issue forms:
      </div>
      <div
        className="space-10"
      />
      <div
        className="box-contact"
      >
        <div
          className="contact-text2"
        >
          This page is for reporting any issues you may have with either the website, or the Learning Spaces themselves.
        </div>
        <div
          className="contact-text2"
        >
          You can report about:
        </div>
        <div
          className="space-3"
        />
        <div
          className="contact-text2"
        >
          - Learning spaces: Participants may have issues with the learning spaces themselves, such as technical issues with the platform or difficulty accessing certain resources. Creating an issue can help bring these problems to the attention of the team responsible for managing the learning space, so that they can be addressed and resolved.
        </div>
        <div
          className="space-3"
        />
        <div
          className="contact-text2"
        >
          - Comments or posts: Participants may have concerns about comments or posts made by other users in the learning space. These could be inappropriate or offensive comments, or posts that are not relevant to the topic being discussed.
        </div>
        <div
          className="space-3"
        />
        <div
          className="contact-text2"
        >
          - UI: Participants may have suggestions or issues with the user interface (UI) of the learning space. They may find certain features confusing or hard to use, or have suggestions for improvements that could make the learning space more user-friendly.                         
        </div>
      </div>
    </div>
    <div
      className="contact-page-middle"
    >
      <div
        className="space-20"
      />
      <div
        className="space-10"
      />
      <div
        className="contact-title"
      >
        Tell Us About Any Issues
      </div>
      <div
        className="space-20"
      />
      <label
        className="ls-box-lb-option"
      >
         Subject of your issue: 
      </label>
      <div
        className="space-12"
      />
      <label>
        <span
          className="contact-box"
          contentEditable="true"
          onInput={[Function]}
          role="textbox"
          value="nameMessage"
        />
      </label>
      <div
        className="space-40"
      />
      <label
        className="ls-box-lb-option"
      >
         Can you please specify the problem? 
      </label>
      <div
        className="space-12"
      />
      <label>
        <span
          className="contact-box2"
          contentEditable="true"
          onInput={[Function]}
          role="textbox"
          value="nameMessage"
        />
      </label>
      <div
        className="space-5"
      />
      <label>
        <input
          className="ls-create-btn"
          onClick={[Function]}
          type="submit"
          value="Submit Issue"
        />
      </label>
    </div>
    <div
      className="contact-page-right"
    >
      <div
        className="space-100"
      />
      <div
        className="left-aligned-img"
      >
        <img
          alt="Contact Page Illustration"
          height={250}
          src="leutee.png"
        />
      </div>
      <div
        className="contact-text2"
      >
        "It is important for our co-learners to have a good and respectful experience in order to feel valued, motivated, and supported in their learning journey."
      </div>
      <div
        className="space-8"
      />
      <div
        className="contact-text3"
      >
        - Team Learnify.
      </div>
    </div>
  </div>
  <div
    className="footer-parent-layout"
    data-testid="footer"
  >
    <div
      className="footer-container"
    >
      <div
        className="footer-row"
      >
        <div
          className="footer-column-l"
        >
          <div
            className="footer-item-bold"
          >
            Learnify
          </div>
          <div
            className="footer-item-reg"
          >
            is a learning space for everyone, where learning is 
            <br />
             achieved through collaboration.
          </div>
        </div>
        <div
          className="footer-column-r"
        >
          <a
            className="footer-item-bold"
            href="/home"
          >
            Home
          </a>
          <a
            className="footer-item-bold"
            href="/categories"
          >
            Categories
          </a>
          <a
            className="footer-item-bold"
            href="/notifications"
          >
            Notifications
          </a>
        </div>
        <div
          className="footer-column-r"
        >
          <a
            className="footer-item-bold"
            href="/profile"
          >
            Profile
          </a>
          <a
            className="footer-item-bold"
            href="/contact"
          >
            Contact
          </a>
          <a
            className="footer-item-bold"
            href="/about"
          >
            About
          </a>
        </div>
        <div
          className="footer-column-r"
        >
          <a
            className="footer-item-bold"
            href="/user/learning-spaces"
          >
            My Learning Spaces
          </a>
          <a
            className="footer-item-bold"
            href="/popular-learning-spaces"
          >
            Popular Learning Spaces
          </a>
          <a
            className="footer-item-bold"
            href="/recommended-learning-spaces"
          >
            Recommended Learning Spaces
          </a>
        </div>
        <div
          className="footer-column-r"
        >
          2022
           Â© Learnify Team. All rights reserved.
        </div>
      </div>
    </div>
  </div>
</div>
`;
